terraform-aws-static-site/main.tf

```
@@ -0,0 +1,163 @@
+# Non www-bucket
+resource "aws_s3_bucket" "b" {
+  bucket = var.bucket_name
+  acl    = "public-read"
+
+  website {
+    index_document = "index.html"
+    error_document = "index.html"
+  }
+}
+
+data "aws_iam_policy_document" "b" {
+  statement {
+    sid    = "PublicReadGetObject"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    actions   = ["s3:GetObject"]
+    resources = ["arn:aws:s3:::${var.bucket_name}/*"]
+  }
+}
```

There are a couple of things we can tighten up permissions-wise here. Rather than having the identifiers as "*", we can specify the cloudfront distribution. This could become something like:

```suggestion

  website {
    index_document = "index.html"
    error_document = "index.html"
  }
}

data "aws_iam_policy_document" "b" {
  statement {
    effect = "Allow"

    principals {
      type        = "AWS"
      identifiers = [aws_cloudfront_origin_access_identity.origin_access_identity.iam_arn]
    }

    actions   = ["s3:GetObject"]
    resources = ["arn:aws:s3:::${var.bucket_name}/*"]
  }
}
```